dts/bindings/input/gpio-kbd-matrix.yaml

```
@@ -0,0 +1,42 @@
+# Copyright 2023 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  GPIO based keyboard matrix input device
+
+  Implement an input device for a GPIO based keyboard matrix.
+
+  Example configuration:
+
+  kbd-matrix {
+          compatible = "gpio-kbd-matrix";
+          row-gpios = <&gpio0 0 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>,
+                      <&gpio0 1 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+          col-gpios = <&gpio0 2 GPIO_ACTIVE_LOW>,
+                      <&gpio0 3 GPIO_ACTIVE_LOW>,
+                      <&gpio0 4 GPIO_ACTIVE_LOW>;
+          no-ghostkey-check;
+  };
+
+compatible: "gpio-kbd-matrix"
+
+include:
+  - name: kbd-matrix-common.yaml
+    property-blocklist:
+      - row-size
+      - col-size
+
+properties:
+  row-gpios:
+    type: phandle-array
+    required: true
+    description: |
+      GPIO for the keyboard matrix rows, must match the count in row-size,
+      supports up to 8 rows, all row GPIO pins must have interrupt support.
```

Drop the "match row-size" language.
```suggestion
      GPIO for the keyboard matrix rows. All row GPIO pins must have interrupt support.
```

---

Done, I kept the note about the limit of 8 as that still applies, it's enforced in the common code.